src/syntax/read.rkt

```diff
@@ -51,29 +58,51 @@
        [(list (cons prop val) rest ...)
         (loop (dict-set* prop-dict prop val) rest)])))
   
-  (define default-repr (get-representation (dict-ref prop-dict* ':precision 'binary64)))
+  (define default-prec (dict-ref prop-dict* ':precision 'binary64))
+  (define default-repr (get-representation default-prec))
   (define var-reprs 
     (for/list ([arg args] [arg-name arg-names])
       (cons arg-name
             (if (and (list? arg) (set-member? args ':precision))
                 (get-representation (list-ref args (add1 (index-of args ':precision))))
                 default-repr))))
 
-  (define body* (desugar-program body default-repr var-reprs))
+  ;; Named fpcores need to be added to function table
+  (when func-name
+    (hash-set! (*functions*) func-name
+               (list args default-prec body)))
```

Comment:
Ugh not thrilled about having yet another place where we store repr names instead of reprs!

Reply:
Might also be good to wrap this `hash-set!` in a function. `(*functions*)` isn't local data, better not to modify it willy-nilly.